### ISSUE TITLE
Firestore persistence isAvailable() check should exclude iOS Safari, Android 4.4, etc.

### ISSUE THREAD

**Issue by nathanfan46** [firebase/firebase-js-sdk#443]
### [REQUIRED] Describe your environment

  * Operating System version: iOS 9.3.5
  * Firebase SDK version: 4.8.0
  * Firebase Product: firestore

### [REQUIRED] Describe the problem
When enable persistence, iPad mini keep showing following error messages:
![unadjustednonraw_thumb_2](https://user-images.githubusercontent.com/914545/35009775-a57ed53a-face-11e7-917c-d54ee8be659b.jpg)

#### Steps to reproduce:
Just enable persistence

`AngularFirestoreModule.enablePersistence()`

#### Relevant Code:

```javascript
// TODO(you): code here to reproduce the problem
```

**Comment by nikhilag** [comment]
I have been seeing the same issue for a while now. Is there any fix planned?

**Comment by jshcrowthe** [comment]
@nathanfan46 Can you get us some code, or a sample repo to illustrate the problem?

**Comment by mikelehen** [comment]
We rely on IndexedDB which is unfortunately not available in Safari 9.  The right "fix" here is to expand our isAvailable() check here [https://github.com/firebase/firebase-js-sdk/blob/master/packages/firestore/src/local/simple_db.ts#L85] to detect Safari 9 so we gracefully fail to enable persistence instead of crashing...  We'll take a look at making this change in an upcoming release.

**Comment by nathanfan46** [comment]
@jshcrowthe I created a simple ionic project with AngularFirestoreModule.enablePersistence()
you can reproduce the issue by running command `ionic cordova run ios -lcs`
https://github.com/nathanfan46/rttest1

@mikelehen Thank you for the response, it would be better to not having persistence than crash.

**Comment by paulstelzer** [comment]
I can confirm the same issue on Android 4.4 (Android 5 and above all working fine)! If I enable persistence I get "Unhandled Promise rejection" and firestore is not working. I check for IndexedDb and he returns that this is available. Is there maybe another reason why enable persistence causes the issue?

**Comment by sengoontoh** [comment]
following

**Comment by nikhilag** [comment]
@mikelehen : Is this issue fixed? I have not been able to enable offline persistence in my web app due to this issue. Please help.

**Comment by paulstelzer** [comment]
I don't know if this always fits, but you can check for `navigator.serviceWorker`. If this is available, you can activate persistence, otherwise not. This seems to work good for me

```
export function FirestorePersistenceEnabler() {
  console.log('FirestorePersistenceEnabler', navigator.serviceWorker);

  if (navigator.serviceWorker === undefined) {
    return false;
  }
  return true;
}
```

**Comment by mikelehen** [comment]
We haven't fixed this.  PRs welcome.  It should just be a matter of updating this check: https://github.com/firebase/firebase-js-sdk/blob/fe48712fd9f0d1e1cd05bb2b706de5302d5c7d6e/packages/firestore/src/local/simple_db.ts#L132

**Comment by mesqueeb** [comment]
Hello @mikelehn
I have a Cordova-app for Android and iOS.
I need to ship my app **only to Android and iOS versions compatible with Firestore offline support** for Firebase JS SDK.

1. Which Android version and up should I target?
2. Which iOS version and up should I target?

**Comment by mikelehen** [comment]
@mesqueeb Sorry, I don't know anything more than is captured in this github issue.  Your best bet will probably be to test for yourself so you can be sure your app works as desired.

**Comment by thebrianchen** [comment]
@mesqueeb 

A list of supported browsers and OS versions can be found here: https://caniuse.com/#feat=indexeddb

**Comment by thebrianchen** [comment]
Fixed in #1703, which will be included in the next SDK release.

iOS Safari for versions < iOS 10 and Android OS < 4.5 will default to not enabling persistence.
